pkg/plugins/manager/loader/loader.go

```diff
@@ -72,17 +83,48 @@ func (l *Loader) Load(ctx context.Context, src plugins.PluginSource) ([]*plugins
 		}
 		bootstrappedPlugins = append(bootstrappedPlugins, bootstrappedPlugin...)
 	}
+	l.log.Debug("Bootstrapped", "class", src.PluginClass(ctx), "duration", time.Since(st))
 
+	st = time.Now()
 	validatedPlugins := []*plugins.Plugin{}
+	type validateResult struct {
+		bootstrappedPlugin *plugins.Plugin
+		err                error
+	}
+	validateResults := make(chan validateResult, len(bootstrappedPlugins))
+
+	// If the PluginsCDNSyncLoaderEnabled feature is enabled, validate plugins in parallel.
+	// Otherwise, validate plugins sequentially.
+	var limitSize int
+	if l.cfg.Features.PluginsCDNSyncLoaderEnabled && src.PluginClass(ctx) == plugins.ClassCDN {
+		limitSize = min(len(bootstrappedPlugins), concurrencyLimit)
+	} else {
+		limitSize = 1
+	}
+	limit := make(chan struct{}, limitSize)
 	for _, bootstrappedPlugin := range bootstrappedPlugins {
-		err := l.validation.Validate(ctx, bootstrappedPlugin)
-		if err != nil {
-			l.recordError(ctx, bootstrappedPlugin, err)
+		bootstrappedPlugin := bootstrappedPlugin
+		limit <- struct{}{}
+		go func() {
+			err := l.validation.Validate(ctx, bootstrappedPlugin)
+			validateResults <- validateResult{
+				bootstrappedPlugin: bootstrappedPlugin,
+				err:                err,
+			}
+			<-limit
+		}()
```

Comment:
```suggestion
		go func(p *plugins.Plugin) {
			err := l.validation.Validate(ctx, p)
			validateResults <- validateResult{
				bootstrappedPlugin: p,
				err:                err,
			}
			<-limit
		}(bootstrappedPlugin)
```

I think this is a bit cleaner?

Reply:
:+1:  agree, added